apps/admin-ui/src/spa/recurring-reservations/application-rounds/[id]/allocation/modules/applicationRoundAllocation.ts

```diff
@@ -242,98 +313,129 @@ export const getApplicantName = (
-): number => {
-  if (!selection || applicationEventSchedules?.length < 1) {
-    return 0;
+// TODO rename to filter + sort
+export function getSelectedApplicationEvents(
+  aes: ApplicationEventNode[],
+  selection: string[] | null,
+  reservationUnitPk: number
+): ApplicationEventNode[] {
+  if (!selection || !aes || selection.length < 1) {
+    return [];
   }
 
-  const timeSlots = getTimeSlots(applicationEventSchedules);
-  const matches = selection?.filter((n) => timeSlots.includes(n)).length;
-
-  return (matches / selection.length) * 100;
-};
+  const selectionDecoded = selection.map(decodeTimeSlot);
+  const isInRange = (slot: TimeSlot, range: TimeSlot[] | null) => {
+    if (!range) {
+      return false;
+    }
+    if (range.length < 2) {
+      return false;
+    }
+    if (slot.day !== range[0].day) {
+      return false;
+    }
+    if (slot.hour < range[0].hour) {
+      return false;
+    }
+    if (slot.hour >= range[1].hour) {
+      return false;
+    }
+    return true;
+  };
 
-export const getEarliestScheduleStart = (
-  applicationEventSchedules: ApplicationEventScheduleNode[]
-): string | null => {
-  const starts = applicationEventSchedules.map((n) => n.begin).sort();
-  return starts?.length > 0 ? starts[0] : null;
-};
+  const day = selectionDecoded[0].day;
+  const filtered = aes
+    .filter((ae) => {
+      // this makes it impossible to revert decline
+      return ae.status !== ApplicationEventStatusChoice.Declined;
+    })
+    .filter((ae) => {
+      const times =
+        ae.applicationEventSchedules
+          ?.filter((s) => s.day === day)
+          .map(getRequestedTime) ?? [];
+      return times.some(
+        (slot) => slot && selectionDecoded.some((item) => isInRange(item, slot))
+      );
+    })
+    .filter((ae) => {
+      const isAllocatedForToday =
+        ae.applicationEventSchedules?.some((s) => s.allocatedDay) ?? false;
+      const nAllocated =
+        ae.applicationEventSchedules?.filter((s) => s.allocatedDay === day)
+          .length ?? 0;
+      const nRequested = ae.eventsPerWeek ?? 0;
+      if (isAllocatedForToday) {
+        const fa = ae.applicationEventSchedules?.filter(
+          (s) => s.allocatedDay === day
+        );
+        if (fa != null && fa?.length > 0) {
+          const fa_ = fa.map((s) => {
+            const timeRange = getAllocatedTime(s);
+            if (!timeRange) {
+              return false;
+            }
+            return selectionDecoded
+              .map((x) => isInRange(x, timeRange))
+              .some((b) => b);
+          });
+          return fa_.some((b) => b);
+        }
+        return nAllocated < nRequested;
+      }
+      return true;
+    })
+    .map((ae) => ({
+      ...ae,
+      applicationEventSchedules:
+        ae.applicationEventSchedules?.filter(
+          (s) =>
+            s.allocatedReservationUnit == null ||
+            s.allocatedReservationUnit.pk === reservationUnitPk
+        ) ?? [],
```

Comment:
TODO comment: why are we hiding allocated here?
because on the right hand side we can't do anything with them (they are shown on the left) but they don't affect the allocation calendar, and no actions are allowed on this page for them